pyrsistent/_pmap.py

```diff
@@ -140,7 +140,9 @@ def __str__(self):
 
     def __hash__(self):
         # This hashing algorithm is probably not the speediest
```

Comment:
We could do some sort of global cache that keys off of ids and uses weakrefs to invalidate the cache maybe... seems like a lot of work and complexity for semantically the same thing.

By the way, this and a few other optimizations we'll be sending upstream soon come from profiling ClusterHQ/flocker. If you are ever interested in seeing some of our profiles or want to know some specific circumstances that we've found slowness within pyrsistent please feel free to reach out. In the past week we've had some pretty interesting findings, especially with large recursively pyrsistent structures.